src/infra/core/src/ingest/fetch_service/mod.rs

```diff
@@ -0,0 +1,407 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::borrow::Cow;
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+use chrono::{DateTime, TimeZone, Utc};
+use container_runtime::*;
+use internal_error::{ErrorIntoInternal, ResultIntoInternal};
+use kamu_core::*;
+use kamu_datasets::{DatasetEnvVar, DatasetKeyValueService};
+use opendatafabric::*;
+use url::Url;
+
+use super::*;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+mod configs;
+pub use configs::*;
+
+mod container;
+#[cfg(feature = "ingest-evm")]
+mod evm;
+mod file;
+#[cfg(feature = "ingest-ftp")]
+mod ftp;
+mod http;
+#[cfg(feature = "ingest-mqtt")]
+mod mqtt;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub const ODF_BATCH_SIZE: &str = "ODF_BATCH_SIZE";
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct FetchService {
+    container_runtime: Arc<ContainerRuntime>,
+    dataset_key_value_svc: Arc<dyn DatasetKeyValueService>,
+    run_info_dir: Arc<RunInfoDir>,
+
+    source_config: Arc<SourceConfig>,
+    http_source_config: Arc<HttpSourceConfig>,
+
+    #[cfg_attr(not(feature = "ingest-evm"), allow(dead_code))]
+    eth_source_config: Arc<EthereumSourceConfig>,
+
+    #[cfg_attr(not(feature = "ingest-mqtt"), allow(dead_code))]
+    mqtt_source_config: Arc<MqttSourceConfig>,
+}
```

Comment:
I definitely do not like when `mod.rs` contains anything but module or macro declarations.

Reply:
As we discussed, I'm also not a fan of code in `mod.rs`, and did this to allow other files to access private functions and types in `mod`.

I refactored to move all main logic from `mod.rs` into `fetch_service/core.rs` and made private methods `pub(super)` (first time using such modifier) to make them accessible to sibling modules.